presto-orc/src/main/java/com/facebook/presto/orc/reader/ListStreamReader.java

```diff
@@ -63,19 +74,32 @@
     private int nextBatchSize;
 
     private InputStreamSource<BooleanInputStream> presentStreamSource = missingStreamSource(BooleanInputStream.class);
-    @Nullable
-    private BooleanInputStream presentStream;
 
     private InputStreamSource<LongInputStream> lengthStreamSource = missingStreamSource(LongInputStream.class);
     @Nullable
     private LongInputStream lengthStream;
 
-    private boolean rowGroupOpen;
-
     // The set of subscripts for which data needs to be
     // returned. Other positions can be initialized to null. If this
     // is null, values for all subscripts must be returned.
-    private int[] subscripts;
+    int[] subscripts;
```

Comment:
subscripts variable is not used - does it mean that pruning doesn't happen?

Also, `SELECT a[1] FROM lineitem_aria CROSS JOIN UNNEST (order_part_supp_array) as t(a)` fails in setReferencedSubfields.